Scripts/Exile's Reach: Ship both factions

malcrom opened the pull request:
Scripting to get Alliance and Horde tunes from ship to the shore.

**Changes proposed:**

-  Add missing scripting to get player to shore. Will continue scripting quest on shore after this PR is added.

**Tests performed:**
Builds, tested both sides in game.

**Known issues and TODO list:** (add/remove lines as needed)

No known issues

Commit:
Scripts/Exile's Reach: Ship

Scripting to get Alliance and Horde tunes from ship to the shore.

Commit:
Scripts/Exile's Reach: Ship

Scripting to get Alliance and Horde tunes from ship to the shore.

Commit:
Scripts/Exile's Reach: Ship

Scripting to get Alliance and Horde tunes from ship to the shore.

Naddley commented on sql/updates/world/master/Exiles_reach.sql:
```diff
@@ -0,0 +1,499 @@
+-- *** Creature Fixes ***
+
+-- Phase updates
+UPDATE `creature` SET `PhaseId`=15286 WHERE `id` IN (166573,166824); -- Warlord Breka Grimaxe
```
There is no PhaseShift after complete the first Quest so they are not phased.

malcrom replied:
Changed

Commit:
Update Exiles_reach.sql

Set CGUID very high to remove
ci/circleci: codestyle_and_sql error

Commit:
Update Exiles_reach.sql

Set CGUID very high to remove
ci/circleci: codestyle_and_sql error

Commit:
Update Exiles_reach.sql

Set CGUID very high to remove
ci/circleci: codestyle_and_sql error

Commit:
Merge branch 'Exiles_Reach' of https://github.com/malcrom/TrinityCore into Exiles_Reach

Commit:
SetStringId changes

Updates suggested by Seydon Tested in game.

Commit:
Merge branch 'Exiles_Reach' of https://github.com/malcrom/TrinityCore into Exiles_Reach

Commit:
Update zone_exiles_reach.cpp

Commit:
Update zone_exiles_reach.cpp

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Update zone_exiles_reach.cpp

Added DamageDealt to prevent damage to player if player is below 90% hp. Added check to despawn if player logs off. This only works if there is another player present. Needs more work.

Needs issue with no summoner guid applied to new sparing partner spawn on player relog and unk_142 target for summon spell needs handling.

Commit:
Update zone_exiles_reach.cpp

Commit:
Update zone_exiles_reach.cpp

Commit:
Update zone_exiles_reach.cpp

remove bad despawn code. Added void ConversationWithPlayer

Commit:
Update zone_exiles_reach.cpp

Some code cleanup

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Merge branch 'master' into Exiles_Reach

malcrom commented:
Only two core issues left.

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Fix sparing partner spawning location and cleanup

Commit:
More updates

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Update zone_exiles_reach.cpp

Commit:
Update zone_exiles_reach.cpp

Commit:
Update zone_exiles_reach.cpp

Commit:
Update scriptnames

Commit:
Revert "Merge branch 'master' into Exiles_Reach"

This reverts commit c31faae0adcc80cfa080523f0176afcb775262fb, reversing
changes made to 15c1bca13a977a182dcb9a0d56dff30b4223fbcf.

Seyden committed:
WIP: Cleanup and simplify

Seyden committed:
Whitespaces

Commit:
Merge branch 'Exiles_Reach' of https://github.com/malcrom/TrinityCore into Exiles_Reach

Commit:
Update zone_exiles_reach.cpp

Commit:
Fix German made errors

Commit:
Update zone_exiles_reach.cpp

Commit:
Revert "Revert "Merge branch 'master' into Exiles_Reach""

This reverts commit 324fa0cf71b900e057d37c05b9193020bded382c.

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Update Exiles_reach.sql

Remove Breka Grimaxe phasing

Commit:
class ExilesReach not working from spell script

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
No more hacks

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Make stringId search range adjustable

malcrom commented:
This PR is complete.

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Merge branch 'master' into Exiles_Reach

mdX7 committed:
Merge remote-tracking branch 'remotes/origin/master' into malcrom/Exiles_Reach

mdX7 committed:
prepare merge 1

mdX7 committed:
reorder scripts by quest

mdX7 committed:
more reordering

mdX7 committed:
remove dropped sql stuff

mdX7 committed:
implemented stand your ground back jump

mdX7 committed:
fix private cole being attackable too early

mdX7 committed:
despawn cole/throg for replace

mdX7 committed:
improved condition naming and dropped one condition (unused phase)

mdX7 committed:
added missing phase shifts

mdX7 committed:
prepare merge

Commit:
Merge branch 'master' into Exiles_Reach

Commit:
Update zone_exiles_reach.cpp

mdX7 committed:
fix uninitialized var

mdX7 committed:
prepare merge

mdX7 committed:
prepare merge
